stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerView.kt

```diff
@@ -0,0 +1,195 @@
+package io.getstream.chat.android.ui.message.composer
+
+import android.content.Context
+import android.text.Editable
+import android.util.AttributeSet
+import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import com.google.android.material.internal.TextWatcherAdapter
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.common.state.MessageAction
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflater
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultCenterContentBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultLeadingContentBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultTrailingContentBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@ExperimentalStreamChatApi
+public class MessageComposerView : ConstraintLayout {
+
+    private lateinit var binding: StreamUiMessageComposerBinding
+
+    /**
+     * @property messageInputState The current state of the [MessageComposerView] distributed using [Flow] API.
+     *
+     * Whenever the value of this property changes updated the [MessageComposerView] re-renders automatically.
+     * If you are using custom [View] implementation for [leadingContent], [centerContent], or [trailingContent] you may
+     * want to collect [messageInputState] items to update them when the state changes.
+     */
+    public val messageInputState: MutableStateFlow<MessageInputState> = MutableStateFlow(MessageInputState())
```

Comment:
Should this potentially be within the ViewModel? I know Views commonly have state they expose, but if users want to access it, they should be able to do it through the ViewModel.

Also - we could expose something like this through the `Controller` instead and just observe it here.

Reply:
Yes, this would be good. Default content definition would land `MessageComposerViewModelBinding` then `MessageComposerView` implementation would be cleaner.

Reply:
I've refactored `messageInputState` in the 8dfa367 commit 👇🏼 
* `messageInputState` is now exposed by `MessageComposerViewModel.messageInputState: StateFlow<MessageInputState>`
* Default content is defined inside `MessageComposerViewModel::bindView` 
* `MessageComposerView` is stateless now